dace/transformation/interstate/condition_map_interchange.py

```diff
@@ -0,0 +1,225 @@
+# Copyright 2019-2025 ETH Zurich and the DaCe authors. All rights reserved.
+"""Interchange conditional blocks with nested map regions."""
+
+from dace import sdfg as sd
+from dace.sdfg import utils as sdutil
+from dace.sdfg.state import ControlFlowRegion, ConditionalBlock
+from dace.sdfg.nodes import MapEntry, MapExit, NestedSDFG
+from dace.memlet import Memlet
+from dace.transformation import transformation
+from dace.sdfg.sdfg import InterstateEdge
+from dace.sdfg.utils import set_nested_sdfg_parent_references
+import copy
+
+
+@transformation.explicit_cf_compatible
+class ConditionMapInterchange(transformation.MultiStateTransformation):
+    """
+    If one or multiple maps are surrounded by a conditional block, moves the conditional block into the nested SDFG of maps.
+    """
+
+    cond_block = transformation.PatternNode(ConditionalBlock)
+
+    @classmethod
+    def expressions(cls):
+        return [sdutil.node_path_graph(cls.cond_block)]
+
+    def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
```

Comment:
Shouldn't there be an additional check to prevent interchanging in the case of a dependency preventing it?
Examples of a dependency I can think of would be:
- based on a symbol:
```Python
if i < 10:
  for i in map[0:10]:
    A[i] += 1
```
- or based on a data dependency:
```Python
if A[4] < 10:
  for i in map[0:10]:
    A[i] += 1
```